pkg/cluster/cluster.go

```diff
@@ -29,6 +31,17 @@ type Cluster struct {
 // handles starting and writing/reading bootstrap data, and returns a channel
 // that will be closed when datastore is ready.
 func (c *Cluster) Start(ctx context.Context) (<-chan struct{}, error) {
+	// check if we need to perform an on-demand etcd snapshot. If we do,
+	// perform the snapshot and exit accordingly.
+	if c.config.EtcdSnapshotNow {
+		if err := c.Snapshot(ctx, c.config); err != nil {
+			logrus.Error(err)
+			os.Exit(1)
+		} else {
+			os.Exit(0)
+		}
```

Comment:
I think Go style guide usually drops a trailing else:
```suggestion
		}
		os.Exit(0)
```

Reply:
Yeah, it does. Remnants of previous edits. Nice catch.